src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -314,51 +344,57 @@ func (sm *StorageMinerActor_I) SubmitPoSt(postSubmission poster.PoStSubmission)
 	// SectorRecovering is Proving and hence will not be in GetZeros()
 	// heavy penalty if Failing for more than or equal to MAX_CONSECUTIVE_FAULTS
 	// otherwise increment FaultCount in Sectors().State
-	for _, sectorNo := range sm.ProvingSet_.SectorsOff() {
-		state, found := sm.Sectors()[sectorNo]
+	for _, sectorNo := range st.Impl().ProvingSet_.SectorsOff() {
+		sectorState, found := st.Sectors()[sectorNo]
 		if !found {
 			continue
 		}
-		switch state.State().StateNumber {
+		switch sectorState.State().StateNumber {
 		case SectorFailingSN:
-			sm.failSector(sectorNo, true)
+			st._updateFailSector(rt, sectorNo, true)
 		default:
-			// TODO: proper failure
-			panic("Invalid sector state in ProvingSet.SectorsOff")
+			// TODO: determine proper error here and error-handling machinery
+			rt.Fatal("Invalid sector state in ProvingSet.SectorsOff")
 		}
 	}
 
 	// Process Expiration.
-	sm.expireSectors()
+	st._updateExpireSectors(rt)
 
-	sm.submitFaultReport(
+	h.Commit(st)
+
+	h, st = a.State(rt)
+	terminationFaultCount := st.SectorTable().Impl().TerminationFaultCount_
+	h.Release(st)
+
+	a._submitFaultReport(
+		rt,
 		util.UVarint(0), // NewDeclaredFaults
 		util.UVarint(0), // NewDetectedFaults
-		util.UVarint(sm.SectorTable().Impl().TerminationFaultCount_),
+		util.UVarint(terminationFaultCount),
 	)
 
-	sm.submitPowerReport()
+	a._submitPowerReport(rt)
 
 	// TODO: check EnsurePledgeCollateralSatisfied
 	// pledgeCollateralSatisfied
 
 	// Reset Proving Period and report power updates
 	// sm.ProvingPeriodEnd_ = PROVING_PERIOD_TIME
 
-	// TODO: get currEpoch from runtime
-	var currEpoch block.ChainEpoch
-	sm.ChallengeStatus().Impl().OnChallengeResponse(currEpoch)
-	sm.processStagedCommittedSectors()
+	h, st = a.State(rt)
+	st.ChallengeStatus().Impl().OnChallengeResponse(rt.CurrEpoch())
+	st._updateCommittedSectors(rt)
```

Comment:
are we missing `h.Commit(st)` here?

Reply:
Yes, good catch -- just fixed in latest commit.